src/main/java/uk/gov/hmcts/reform/migration/service/RespondentsDataMigrationService.java

```diff
@@ -0,0 +1,149 @@
+            .collect(toList());
+
+        // ADD FIRST RESPONDENT TO ADDITIONAL RESPONDENT LIST
+
+        migratedRespondentCollection.add(migratedFirstRespondent);
+
+        List<Map<String, Object>> newStructure = new ArrayList<>();
+
+        /// BUILD NEW STRUCTURE
+
+        migratedRespondentCollection.forEach(item -> {
+            Map map = objectMapper.convertValue(item, Map.class);
+
+            newStructure.add(ImmutableMap.of(
+                "id", UUID.randomUUID().toString(),
+                "value", map));
+        });
+
+        log.info("returning new structure {}", newStructure);
+
+        return newStructure;
+    }
+
+    private Respondent migrateIndividualRespondent(OldRespondent or) {
+        log.info("migrating respondent {}", or);
+
+        Address.AddressBuilder addressBuilder = Address.builder();
+        if (!isEmpty(or.getAddress())) {
+            addressBuilder.addressLine1(defaultIfBlank(or.getAddress().getAddressLine1(), null));
+            addressBuilder.addressLine2(defaultIfBlank(or.getAddress().getAddressLine2(), null));
+            addressBuilder.addressLine3(defaultIfBlank(or.getAddress().getAddressLine3(), null));
+            addressBuilder.postTown(defaultIfBlank(or.getAddress().getPostTown(), null));
+            addressBuilder.postcode(defaultIfBlank(or.getAddress().getPostcode(), null));
+            addressBuilder.county(defaultIfBlank(or.getAddress().getCounty(), null));
+            addressBuilder.country(defaultIfBlank(or.getAddress().getCountry(), null));
+        }
+        Address address = addressBuilder.build();
+
+        TelephoneNumber.TelephoneNumberBuilder telephoneNumberBuilder = TelephoneNumber.builder();
+        telephoneNumberBuilder.telephoneNumber(defaultIfBlank(or.getTelephone(), null));
+        TelephoneNumber telephoneNumber = telephoneNumberBuilder.build();
+
+        RespondentParty.RespondentPartyBuilder partyBuilder = RespondentParty.builder();
+        partyBuilder.partyID(UUID.randomUUID().toString());
+
+        if (!isEmpty(or.getName())) {
+            partyBuilder.firstName(splitName(or.getName()).get(0));
+
+            if (splitName(or.getName()).size() > 1) {
+                partyBuilder.lastName(splitName(or.getName()).get(1));
+            }
+        }
+
+        partyBuilder.dateOfBirth(defaultIfBlank(or.getDob(), null));
+        partyBuilder.address(address);
+        partyBuilder.telephoneNumber(telephoneNumber);
+        partyBuilder.gender(defaultIfBlank(or.getGender(), null));
+        partyBuilder.genderIdentification(defaultIfBlank(or.getGenderIdentify(), null));
+        partyBuilder.placeOfBirth(defaultIfBlank(or.getPlaceOfBirth(), null));
+        partyBuilder.relationshipToChild(defaultIfBlank(or.getRelationshipToChild(), null));
+        partyBuilder.contactDetailsHidden(defaultIfBlank(or.getContactDetailsHidden(), null));
```

Comment:
One thing is missing. We capture reason for contact details hide. We need to add it otherwise we lose it.

Reply:
good spot 👍